pkg/backend/openshiftcluster.go

```diff
@@ -351,11 +353,22 @@ func (ocb *openShiftClusterBackend) asyncOperationResultLog(log *logrus.Entry, i
 			"properties.servicePrincipalProfile", "The Azure Red Hat Openshift resource provider service principal has been removed from your tenant. To restore, please unregister and then re-register the Azure Red Hat OpenShift resource provider.")
 	}
 
-	err, ok := backendErr.(*api.CloudError)
-	if ok {
-		resultType := utillog.MapStatusCodeToResultType(err.StatusCode)
-		log = log.WithField("resultType", resultType)
+	var statusCode int
 
+	// Get the HTTP status code and map it to a result type
+	if err, ok := backendErr.(*api.CloudError); ok {
```

Comment:
Sorry I oversaw these error handlings.
If there's no reason to use type assertion, please use `errors.Is` or `errors.As` instead?
https://issues.redhat.com/browse/ARO-7922

Reply:
@bitoku No worries, there isn't :smile: , it's just old ways. Will update this.